src/recommendpolicy/helperFunctions.go

```diff
@@ -189,3 +193,93 @@ func initDeploymentWatcher() {
 		}
 	}
 }
+
+func mergePolicies(policies []types.KnoxSystemPolicy, ms types.MatchSpec, name, namespace string, labels LabelMap) types.KnoxSystemPolicy {
+
+	var merged types.KnoxSystemPolicy
+
+	merged.APIVersion = "v1"
+	merged.Kind = "KubeArmorPolicy"
+	merged.Metadata = map[string]string{
+		"name":      fmt.Sprintf("%v-%v", types.HardeningMergePolicy, name),
+		"namespace": namespace,
+	}
+	merged.Spec.Selector.MatchLabels = labels
+
+	for _, policy := range policies {
+		for i := range policy.Spec.File.MatchDirectories {
+			dir := policy.Spec.File.MatchDirectories[i]
+			dir.Action = policy.Spec.Action
+			dir.Severity = policy.Spec.Severity
+			dir.Message = policy.Spec.Message
+			dir.Tags = policy.Spec.Tags
+			merged.Spec.File.MatchDirectories = append(merged.Spec.File.MatchDirectories, dir)
+		}
+	}
+
+	for _, policy := range policies {
+		for i := range policy.Spec.File.MatchPaths {
+			path := policy.Spec.File.MatchPaths[i]
+			path.Action = policy.Spec.Action
+			path.Severity = policy.Spec.Severity
+			path.Message = policy.Spec.Message
+			path.Tags = policy.Spec.Tags
+			merged.Spec.File.MatchPaths = append(merged.Spec.File.MatchPaths, path)
+		}
+	}
+
+	for _, policy := range policies {
+		for i := range policy.Spec.Process.MatchDirectories {
+			dir := policy.Spec.Process.MatchDirectories[i]
+			dir.Action = policy.Spec.Action
+			dir.Severity = policy.Spec.Severity
+			dir.Message = policy.Spec.Message
+			dir.Tags = policy.Spec.Tags
+			merged.Spec.Process.MatchDirectories = append(merged.Spec.Process.MatchDirectories, dir)
+		}
+	}
+
+	for _, policy := range policies {
+		for i := range policy.Spec.Process.MatchPaths {
+			path := policy.Spec.Process.MatchPaths[i]
+			path.Action = policy.Spec.Action
+			path.Severity = policy.Spec.Severity
+			path.Message = policy.Spec.Message
+			path.Tags = policy.Spec.Tags
+			merged.Spec.Process.MatchPaths = append(merged.Spec.Process.MatchPaths, path)
+		}
+	}
+
+	return merged
+}
+
+func GroupPoliciesByLabel(policies []types.KnoxSystemPolicy, ms types.MatchSpec, name, namespace string, labels LabelMap) []types.KnoxSystemPolicy {
```

Comment:
As per my understanding this function is not required. This is just a complicated step to make sure that we are only merging policies from a single namespace with label `x=y`, which is not required.